core/datainput.gms

```diff
@@ -790,89 +790,63 @@ $endif.Base_Cprice
 display pm_regiEarlyRetiRate;
 
 ***---------------------------------------------------------------------------
-*RP* calculate omegs and opTimeYr2te
+*** Calculate lifetime parameters (omeg and opTimeYr2te)
 ***---------------------------------------------------------------------------
-*RP* use new lifetimes defined in generisdata_tech.prn:
-pm_omeg(regi,opTimeYr,te) = 0;
 
 *** FS: use lifetime of tdh2s for tdh2b and tdh2i technologies
 *** which are only helper technologies for consistent H2 use in industry and buildings
 pm_data(regi,"lifetime","tdh2i") = pm_data(regi,"lifetime","tdh2s");
 pm_data(regi,"lifetime","tdh2b") = pm_data(regi,"lifetime","tdh2s");
 
-loop(regi,
-        p_aux_lifetime(regi,te) = 5/4 * pm_data(regi,"lifetime",te);
-        loop(te,
-
-                loop(opTimeYr,
-                        pm_omeg(regi,opTimeYr,te) = 1 - ((opTimeYr.val-0.5) / p_aux_lifetime(regi,te))**4 ;
-                        opTimeYr2te(te,opTimeYr)$(pm_omeg(regi,opTimeYr,te) > 0 ) =  yes;
-                        if( pm_omeg(regi,opTimeYr,te) <= 0,
-                                pm_omeg(regi,opTimeYr,te) = 0;
-                                opTimeYr2te(te,opTimeYr) =  no;
-                        );
-                )
-        );
-);
-
-*** calculate mapping tsu2opTimeYr
-alias(ttot, tttot);
-tsu2opTimeYr(ttot,opTimeYr) =  no;
-tsu2opTimeYr(ttot,"1") =  yes;
-loop(ttot,
-   loop(opTimeYr,
-      loop(tttot $(ord(tttot) le ord(ttot)),
-         if(opTimeYr.val = pm_ttot_val(ttot)-pm_ttot_val(tttot)+1,
-            tsu2opTimeYr(ttot,opTimeYr) =  yes;
-         );
-      );
-   );
+*** Compute the depreciation of technologies over their lifetime
+*' Technologies depreciate over their lifetime.
+*' Their remaining capacity pm_omeg starts at 1 and decreases toward zero with a curve of exponent 4:
+*' slow depreciation during the first half of the lifetime and faster during the second half.
```

Comment:
add maybe here "When the curve of capacity vs. time hits y = 0 (x axis), x= 5/4. So there is still some capacity beyond the age of the plant, but the total area under the curve equals capacity * lifetime"

..if my understanding is correct here

Reply:
I would write "beyond the average technical lifetime of the technology as stated in `lifetime`, but the total ..." , otherwise I agree this could help future REMIND users understand what is done here